Web/Exercise/04/script.js

```diff
@@ -0,0 +1,27 @@
+const user = {
+    name: "Piyush Sharma",
+    designation: "Senior Software Engineer",
+    company: "Infuse Consulting",
+    hobbies: ["Reading", "Listening to music", "Collecting stamps"]
+}
+
+const printUserProfile = ({name, designation, company, hobbies}) => {
+    let str = "";
+    if (hobbies.length == 1) {
+        str = hobbies[0];
+    }
+    else {
+        for (let i = 0; i < hobbies.length; i++) {
+            if (i == hobbies.length - 1) {
+                str += "and " + hobbies[i];
+            }
+            else {
+                str = hobbies[i] + ", "
+            }
+        }
+    }
```

Comment:
good try! you can enhance it using array's `reduce` method if you're interested.